h2/src/main/org/h2/mvstore/MVStore.java

```diff
@@ -370,7 +370,12 @@ to a map (possibly the metadata map) -
         if (this.fileStore != null) {
             retentionTime = this.fileStore.getDefaultRetentionTime();
             // 19 KB memory is about 1 KB storage
-            int kb = Math.max(1, Math.min(19, Utils.scaleForAvailableMemory(64))) * 1024;
+            int kb;
+            try {
+               kb = Math.max(1, Math.min(19, Utils.scaleForAvailableMemory(64))) * 1024;
+            } catch(NoClassDefFoundError ex) {
+                kb = 19 * 1024;     // this equates to 19MB
+            }
```

Comment:
I think this isn't a proper place to catch this error, the `Utils.scaleForAvailableMemory` method should catch it together with other exceptions. It catches only exceptions, but I think it should catch any `Throwable`.

Reply:
Good point. I hadn't actually looked at Utils.java. Simply changing the catch from Exception to Error might handle it.